hibernate-core/src/main/java/org/hibernate/bytecode/enhance/spi/interceptor/EnhancementHelper.java

```diff
@@ -47,27 +53,89 @@ public static boolean includeInBaseFetchGroup(
 			return true;
 		}
 
+		// if we get here, we know the property owner is enhanced for laziness
+		//
+		// NOTE : we make the (potentially untrue) assumption here that
+		// if the owner is enhanced, then all classes are enhanced..
+
 		if ( value instanceof ToOne ) {
 			final ToOne toOne = (ToOne) value;
-			if ( toOne.isLazy() ) {
-				if ( toOne.isUnwrapProxy() ) {
-					if ( toOne instanceof OneToOne ) {
-						return false;
-					}
-					// include it in the base fetch group so long as the config allows
-					// using the FK to create an "enhancement proxy"
-					//return allowEnhancementAsProxy;
-					// ^^ previously we had to explicitly enable use of enhanced proxies.
-					//		for the moment just return `true` assuming we can.
-					//
-					//		there are cases where this block overall misses quite a few cases
-					//		where it returns the least optimal value.  This is true even outside
-					//		this enhanced-proxy point
-					//
-					//		those will all be addressed in the commits for HHH-13658
-					return true;
+
+			if ( ! toOne.isLazy() ) {
+				// its not lazy... select it
+				return true;
+			}
+
+			// it is lazy.  see if we should select the FK
+
+			if ( bootMapping.getLazyGroup() != null ) {
+				// a non-base fetch group was explicitly specified
+				//
+				// really this should indicate to not select it as part of the base group.
+				// however, at the time being that leads to inefficient SQL - so for now
+				// we simply log a message that we are ignoring the `@LazyGroup` for to-ones
+
+				BytecodeInterceptorLogging.MESSAGE_LOGGER.lazyGroupIgnoredForToOne(
+						bootMapping.getPersistentClass().getEntityName(),
+						bootMapping.getName(),
+						bootMapping.getLazyGroup()
+				);
+
+				// at a later time - for example 6.0 when we can implement the join solution
+				// todo (6.0) : implement this
+				// return false;
+
+				// for now, fall through
+			}
+
+			if ( ! toOne.isReferenceToPrimaryKey() ) {
+				// we do not have a reference to the associated primary-key
+				return false;
+			}
+
+			if ( toOne.getColumnSpan() == 0 ) {
+				// generally this would indicate a "shared PK" on-to-one and there
+				// is no column for the association on the owner table - do not add
+				// the association to the base group (which would force an immediate
+				// select from the association table effectively making this
+				// association non-lazy)
+				return false;
+			}
+
+			final boolean unwrapExplicitlyRequested = toOne.isUnwrapProxy() && !toOne.isUnwrapProxyImplicit();
+
+			if ( inheritanceChecker.hasSubclasses( toOne.getReferencedEntityName() ) ) {
+				// the associated type has subclasses - we cannot use the enhanced proxy and will generate a HibernateProxy
+				if ( unwrapExplicitlyRequested ) {
+					// NO_PROXY was explicitly requested
```

Comment:
I think this is a general confusion.  The return here is not whether we get a proxy or not (nor what kind).  We are simply deciding whether to select the column, here the FK.  

We want to select the FK, so we return true

Reply:
But doesn't that currently result in either a proxy initialized or eager loading of the target entity?

I thought that, by deferring the FK loading, we could avoid this. Or is what I described (proxy initialized or eager loading) not what is happening?

Reply:
if it has subclasses we create a HibernateProxy.  To create a HibernateProxy we need to know the FK

Reply:
Ok, I thought that we could honor the explicitly requested `NO_PROXY` by not selecting the FK, but I'm not 100% sure what the effect of doing that would be.

Reply:
We could, but it would mean selecting the FK separately which is something we try to avoid when we can.  In fact, it would mean 2 selects to initialize the association - 1 to select the FK and then another to select from the association table.  (see why joins would be beneficial?)

Reply:
Ok so in 6.0 we would return false when we support doing that in 1 query?

Reply:
That is one option, yes.  It comes back to the join discussion.  This becomes a very involved discussion as I think there are a lot of conditions that we simply do not account for today such as an explicit `@LazyGroup`, FetchMode#JOIN, etc